QR Codes issue282

easoftwareGit opened the pull request:
<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

<!-- Made suggested changed. Added subjcets.md and tags.md -->

### Type of Change

<!--- Suggested edits and file additions -->

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

<!-- Please check ALL the boxes: -->

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

SSwiniarski commented on content/general/concepts/qr-code/qr-code.md:
```diff
@@ -0,0 +1,39 @@
+---
```
@easoftwareGit The metadata needs to look like this:
```
---
Title: 'The Title'
Description: 'A brief description'
Subjects:
  - 'A subject name'
  - 'A second subject name'
  - 'An nth subject name'
Tags:
  - 'A tag'
  - 'A second tag'
  - 'An nth tag'
CatalogContent:
  - 'learn-example-course'
  - 'path/example-path'
---
```

Subjects come from [this list](https://github.com/Codecademy/docs/blob/main/documentation/subjects).
Tags come from [this list](https://github.com/Codecademy/docs/blob/main/documentation/tags.md).
For catalog content use [this document](https://github.com/Codecademy/docs/blob/main/documentation/catalog-content.md) as a guide.

SSwiniarski commented on content/general/concepts/qr-code/qr-code.md:
```diff
@@ -0,0 +1,39 @@
+
+## What is a QR Code?
+
+A **QR Code** (quick response code) is code that can be read by a computer, just like the standard bar codes that appear on products at retail stores. But the QR Code It is different than the standard bar code. The standard bar code is one dimensional and reads horizontally. A QR Code is a square, with the data enclosed within the square. The three squares in the corners of the QR code tell the Q ...
```
```suggestion
A **QR Code** (quick response code) is code that can be read by a computer, just like the standard bar codes that appear on products at retail stores. But the QR Code is different than the standard bar code. The standard bar code is one dimensional and reads horizontally. A QR Code is a square, with the data enclosed within the square. The three squares in the corners of the QR code tell the QR Code scanning software how the QR Code is orientated, so the scanning software can read the QR Code’s data correctly. [Here](https://www.qr-code-generator.com/qr-code-marketing/qr-codes-basics/) is a good web page detailing how the data is structured in the QR Code.  
```

SSwiniarski commented on content/general/concepts/qr-code/qr-code.md:
```diff
@@ -0,0 +1,39 @@
+
+Software is used to create the QR Code. The user enters data to be encoded and how the QR Code will be used. When the user is done entering the data, the software will create the QR Code. There are several QR Code generator websites. Some will generate basic QR Codes for free.
+
+Some QR Codes are static, and some are dynamic. A static QR Code has all the data encoded, and the data will not change. If a data entry error is made during QR Code creation, that error is permanent. A dynamic QR Code fixes the problem. A dynamic QR	Code will retrieve data from the internet, allowing the QR Code owner to edit the data as needed.
```
```suggestion
Some QR Codes are static, and some are dynamic. A static QR Code has all the data encoded, and the data will not change. If a data entry error is made during QR Code creation, that error is permanent. A dynamic QR Code fixes the problem. A dynamic QR Code will retrieve data from the internet, allowing the QR Code owner to edit the data as needed.
```

SSwiniarski commented on content/general/concepts/qr-code/subjects.md:
```diff
@@ -0,0 +1,4 @@
+General
```
This is not how subjects are added, you need to remove this file from the PR.

SSwiniarski replied:
You still need to remove this file from the PR.

SSwiniarski commented on content/general/concepts/qr-code/tags.md:
```diff
@@ -0,0 +1,5 @@
+Bar Code
```
This is not how tags are added, you need to remove this file from the PR.

SSwiniarski reviewed (changes requested):
@easoftwareGit I did another review of your changes and left a few more comments. You should really read the [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md#metadata) to understand how the metadata section works.

Also, there was no need for a new PR for these changes, you can edit and commit the changes to the branch you're working on and the changes will appear in the PR. I'm going to close the prior PR, and we can work on this one now.

SSwiniarski commented on content/general/concepts/qr-code/qr-code.md:
```diff
@@ -0,0 +1,50 @@
+
+## What is a QR Code?
+
+A **QR Code** (quick response code) is code that can be read by a computer, just like the standard bar codes that appear on products at retail stores. But the QR Code It is different than the standard bar code. The standard bar code is one dimensional and reads horizontally. A QR Code is a square, with the data enclosed within the square. The three squares in the corners of the QR code tell the Q ...
```
```suggestion
A **QR Code** (quick response code) is code that can be read by a computer, just like the standard bar codes that appear on products at retail stores. But the QR Code is different than the standard bar code. The standard bar code is one dimensional and reads horizontally. A QR Code is a square, with the data enclosed within the square. The three squares in the corners of the QR code tell the QR Code scanning software how the QR Code is orientated, so the scanning software can read the QR Code’s data correctly. [Here](https://www.qr-code-generator.com/qr-code-marketing/qr-codes-basics/) is a good web page detailing how the data is structured in the QR Code.  
```

easoftwareGit replied:
I do not see what changes you are proposing here. the "QR Code" appears to have the leading and trailing **, and is blue. Is there something else I am missing?

SSwiniarski replied:
@easoftwareGit: "But the QR Code It is different" --> "But the QR Code is different"

easoftwareGit replied:
Got it. Took me several time sto read it, but finally found the extra It.

SSwiniarski commented on content/general/concepts/qr-code/qr-code.md:
```diff
@@ -0,0 +1,50 @@
+---
+Title: 'QR Codes'
```
@easoftwareGit The metadata needs to look like this:

```
---
Title: 'The Title'
Description: 'A brief description'
Subjects:
  - 'A subject name'
  - 'A second subject name'
  - 'An nth subject name'
Tags:
  - 'A tag'
  - 'A second tag'
  - 'An nth tag'
CatalogContent:
  - 'learn-example-course'
  - 'path/example-path'
---
```

Subjects come from [this list](https://github.com/Codecademy/docs/blob/main/documentation/subjects).
Tags come from [this list](https://github.com/Codecademy/docs/blob/main/documentation/tags.md).
For catalog content use [this document](https://github.com/Codecademy/docs/blob/main/documentation/catalog-content.md) as a guide.

SSwiniarski reviewed (changes requested):
@easoftwareGit you marked my comments as resolved, but you didn't address them. Make sure you're changing the file in the branch associated with this PR, not the old closed one. Let me know if you're having trouble understanding what is required.

easoftwareGit commented:
I am using Visual Studio Code and the GitHub plugin.

I did make the changes, and then marked them as resolved. I moved data from
Subjects.md and tags.md into qr-code.md,  and inserted and formatted the
data as you suggested. I then added subjects.md and tags.md to .gitignore.

I saved the file with changes. I then committed the file

I have now deleted subjects.md and tags.md.

On Mon, Feb 13, 2023 at 9:31 AM SSwiniarski ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> @easoftwareGit <https://github.com/easoftwareGit> you marked my comments
> as resolved, but you didn't address them. Make sure you're changing the
> file in the branch associated with this PR, not the old closed one. Let me
> know if you're having trouble understanding what is required.
> ------------------------------
>
> In content/general/concepts/qr-code/subjects.md
> <https://github.com/Codecademy/docs/pull/1627#discussion_r1104805204>:
>
> > @@ -0,0 +1,4 @@
>
> +General
>
>
> You still need to remove this file from the PR.
> ------------------------------
>
> In content/general/concepts/qr-code/tags.md
> <https://github.com/Codecademy/docs/pull/1627#discussion_r1104805782>:
>
> > @@ -0,0 +1,5 @@
>
> +Bar Code
>
>
> You still need to remove this file from the PR.
> ------------------------------
>
> In content/general/concepts/qr-code/qr-code.md
> <https://github.com/Codecademy/docs/pull/1627#discussion_r1104807107>:
>
> > +Tags:
>
> +  - Bar Code
>
> +  - General
>
> +  - QR Code
>
> +  - Quick Response
>
> +  - Scan
>
> +CatalogContent:
>
> +  - 'General'
>
> +  - 'QR Codes'
>
> +  - 'Subjects'
>
> +  - 'Tags'
>
> +---
>
> +
>
> +## What is a QR Code?
>
> +
>
> +A **QR Code** (quick response code) is code that can be read by a computer, just like the standard bar codes that appear on products at retail stores. But the QR Code It is different than the standard bar code. The standard bar code is one dimensional and reads horizontally. A QR Code is a square, with the data enclosed within the square. The three squares in the corners of the QR code tell the QR Code scanning software how the QR Code is orientated, so the scanning software can read the QR Code’s data correctly. [Here](https://www.qr-code-generator.com/qr-code-marketing/qr-codes-basics/) is a good web page detailing how the data is structured in the QR Code.
>
>
> ⬇️ Suggested change
>
> -A **QR Code** (quick response code) is code that can be read by a computer, just like the standard bar codes that appear on products at retail stores. But the QR Code It is different than the standard bar code. The standard bar code is one dimensional and reads horizontally. A QR Code is a square, with the data enclosed within the square. The three squares in the corners of the QR code tell the QR Code scanning software how the QR Code is orientated, so the scanning software can read the QR Code’s data correctly. [Here](https://www.qr-code-generator.com/qr-code-marketing/qr-codes-basics/) is a good web page detailing how the data is structured in the QR Code.
>
> +A **QR Code** (quick response code) is code that can be read by a computer, just like the standard bar codes that appear on products at retail stores. But the QR Code is different than the standard bar code. The standard bar code is one dimensional and reads horizontally. A QR Code is a square, with the data enclosed within the square. The three squares in the corners of the QR code tell the QR Code scanning software how the QR Code is orientated, so the scanning software can read the QR Code’s data correctly. [Here](https://www.qr-code-generator.com/qr-code-marketing/qr-codes-basics/) is a good web page detailing how the data is structured in the QR Code.
>
>
> ------------------------------
>
> In content/general/concepts/qr-code/qr-code.md
> <https://github.com/Codecademy/docs/pull/1627#discussion_r1104808594>:
>
> > @@ -0,0 +1,50 @@
>
> +---
>
> +Title: 'QR Codes'
>
>
> @easoftwareGit <https://github.com/easoftwareGit> The metadata needs to
> look like this:
>
> ---
>
> Title: 'The Title'
>
> Description: 'A brief description'
>
> Subjects:
>
>   - 'A subject name'
>
>   - 'A second subject name'
>
>   - 'An nth subject name'
>
> Tags:
>
>   - 'A tag'
>
>   - 'A second tag'
>
>   - 'An nth tag'
>
> CatalogContent:
>
>   - 'learn-example-course'
>
>   - 'path/example-path'
>
> ---
>
>
> Subjects come from this list
> <https://github.com/Codecademy/docs/blob/main/documentation/subjects>.
> Tags come from this list
> <https://github.com/Codecademy/docs/blob/main/documentation/tags.md>.
> For catalog content use this document
> <https://github.com/Codecademy/docs/blob/main/documentation/catalog-content.md>
> as a guide.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Codecademy/docs/pull/1627#pullrequestreview-1296130636>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALBP54WOZVARH73AM5VDDVLWXJVVXANCNFSM6AAAAAAUV2K2HM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

SSwiniarski commented:
@easoftwareGit: Now you need to update the metadata. The items under Subjects should only come from the [subjects.md](https://github.com/Codecademy/docs/blob/main/documentation/subjects.md) file in the documentation folder. Tags should come from the [tags.md](https://github.com/Codecademy/docs/blob/main/documentation/tags.md) file in the documentation folder. Please change the Subjects and Tags in the metadata to reflect this. Also, you should _not_ be editing .gitignore, Please undo those edits.

easoftwareGit commented:
Done

On Tue, Feb 14, 2023 at 4:27 AM SSwiniarski ***@***.***>
wrote:

> @easoftwareGit <https://github.com/easoftwareGit>: Now you need to update
> the metadata. The items under Subjects should only come from the
> subjects.md
> <https://github.com/Codecademy/docs/blob/main/documentation/subjects.md>
> file in the documentation folder. Tags should come from the tags.md
> <https://github.com/Codecademy/docs/blob/main/documentation/tags.md> file
> in the documentation folder. Please change the Subjects and Tags in the
> metadata to reflect this. Also, you should *not* be editing .gitignore,
> Please undo those edits.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Codecademy/docs/pull/1627#issuecomment-1429669675>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALBP54S4NJ37434PWQQMBZLWXN23FANCNFSM6AAAAAAUV2K2HM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

SSwiniarski commented:
@easoftwareGit I think you're still not understanding how the metadata works. Let me show you this again:

```
---
Title: 'The Title'
Description: 'A brief description'
Subjects:
  - 'A subject name'
  - 'A second subject name'
  - 'An nth subject name'
Tags:
  - 'A tag'
  - 'A second tag'
  - 'An nth tag'
CatalogContent:
  - 'learn-example-course'
  - 'path/example-path'
---
```
This is how the metadata is supposed to look at the head of your entry. It has a `Subjects:` section and a `Tags:` section. 

There are supposed to be two or three items under each of `Subjects:` and `Tags:`. 

You are supposed to select items that are already listed in the `subjects.md` file, and copy them to your entry under the `Subjects:` section of the metadata. 

The same goes for the `Tags:` section. You look at the items already listed in the `tags.md` file and copy them to your entry under the `Tags:` section of the metadata. 

The `subjects.md` file and the `tags.md` file are there to show you what you can list under the `Subjects:` and `Tags:` section of the Metadata. Most entries do not edit the `subjects.md` file or the `tags.md` file.

While you _can_ add items to the `subjects.md` file, that is extremely rare and what you've added here is not nearly broad enough to count as proper subjects.

You can also add new tags to the `tags.md` file, but you should reduce the redundancy of your added tags, and also list them in the `Tags:` section of the metadata.

Please let me know if this is still unclear to you.

easoftwareGit commented:
Finally it is clicking in. I will go and make the changes.

On Wed, Feb 15, 2023, 9:14 AM SSwiniarski ***@***.***> wrote:

> @easoftwareGit <https://github.com/easoftwareGit> I think you're still
> not understanding how the metadata works. Let me show you this again:
>
> ---
> Title: 'The Title'
> Description: 'A brief description'
> Subjects:
>   - 'A subject name'
>   - 'A second subject name'
>   - 'An nth subject name'
> Tags:
>   - 'A tag'
>   - 'A second tag'
>   - 'An nth tag'
> CatalogContent:
>   - 'learn-example-course'
>   - 'path/example-path'
> ---
>
> This is how the metadata is supposed to look at the head of your entry. It
> has a Subjects: section and a Tags: section.
>
> There are supposed to be two or three items under each of Subjects: and
> Tags:.
>
> You are supposed to select items that are already listed in the
> subjects.md file, and copy them to your entry under the Subjects: section
> of the metadata.
>
> The same goes for the Tags: section. You look at the items already listed
> in the tags.md file and copy them to your entry under the Tags: section
> of the metadata.
>
> The subjects.md file and the tags.md file are there to show you what you
> can list under the Subjects: and Tags: section of the Metadata. Most
> entries do not edit the subjects.md file or the tags.md file.
>
> While you *can* add items to the subjects.md file, that is extremely rare
> and what you've added here is not nearly broad enough to count as proper
> subjects.
>
> You can also add new tags to the tags.md file, but you should reduce the
> redundancy of your added tags, and also list them in the Tags: section of
> the metadata.
>
> Please let me know if this is still unclear to you.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Codecademy/docs/pull/1627#issuecomment-1431714549>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALBP54VYF43SHRXF5HRETETWXUFIJANCNFSM6AAAAAAUV2K2HM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

easoftwareGit commented:
Before I submit changes, here is what I have done:
1) updated Subjects and Tags section at the top of qr-code.md
removed my edits from /documentation/subjects.ms and /documentation/tags.md

Thanks again for your help.

On Wed, Feb 15, 2023 at 9:17 AM ***@***.***> wrote:

> Finally it is clicking in. I will go and make the changes.
>
> On Wed, Feb 15, 2023, 9:14 AM SSwiniarski ***@***.***>
> wrote:
>
>> @easoftwareGit <https://github.com/easoftwareGit> I think you're still
>> not understanding how the metadata works. Let me show you this again:
>>
>> ---
>> Title: 'The Title'
>> Description: 'A brief description'
>> Subjects:
>>   - 'A subject name'
>>   - 'A second subject name'
>>   - 'An nth subject name'
>> Tags:
>>   - 'A tag'
>>   - 'A second tag'
>>   - 'An nth tag'
>> CatalogContent:
>>   - 'learn-example-course'
>>   - 'path/example-path'
>> ---
>>
>> This is how the metadata is supposed to look at the head of your entry.
>> It has a Subjects: section and a Tags: section.
>>
>> There are supposed to be two or three items under each of Subjects: and
>> Tags:.
>>
>> You are supposed to select items that are already listed in the
>> subjects.md file, and copy them to your entry under the Subjects:
>> section of the metadata.
>>
>> The same goes for the Tags: section. You look at the items already
>> listed in the tags.md file and copy them to your entry under the Tags:
>> section of the metadata.
>>
>> The subjects.md file and the tags.md file are there to show you what you
>> can list under the Subjects: and Tags: section of the Metadata. Most
>> entries do not edit the subjects.md file or the tags.md file.
>>
>> While you *can* add items to the subjects.md file, that is extremely
>> rare and what you've added here is not nearly broad enough to count as
>> proper subjects.
>>
>> You can also add new tags to the tags.md file, but you should reduce the
>> redundancy of your added tags, and also list them in the Tags: section
>> of the metadata.
>>
>> Please let me know if this is still unclear to you.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Codecademy/docs/pull/1627#issuecomment-1431714549>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ALBP54VYF43SHRXF5HRETETWXUFIJANCNFSM6AAAAAAUV2K2HM>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>

SSwiniarski commented:
@easoftwareGit so far so good. Now you need to edit the metadata so that the list of `Subjects:` are only from items that appear in the `subjects.md` file, and the items in `Tags:` only come from the `tags.md` file, and yo only use `CatalogContent:` as it appears in the `catalog-content.md`.  Those files exist to show you what is acceptable content for those elements of the metadata.

I want to make sure you understand this before you make any further edits. You use the files in `docs/documentation/` to tell you what you can list in the metadata.

easoftwareGit commented:
OK. Let me say it back to you so we can make sure I am doing it correctly.

For Subjects, Tags and CatalogContent sections in the header of my
qr-code.md file:
I need to scan the matching ____.md file (subjects,md, tags.md or
catalog-content.md), and if there is an item in the ____.md file that I
think should be in the corresponding header section, then I place that word
from the ____.md file into to correct header section in mt qr-codes.md file.

For example: in subjects.md, the first item is "Bash/Shell". That item
should NOT be in my Subjects header section. But the 3rd item "Computer
Science" might be included. "Data Science" is probably NOT included.

So I go through the lists, and pick out the values that I think match my
qr-code.md, and put the matching values in the corresponding sections. If
the item is NOT in the subjects,md, tags.md or catalog-content.md file, it
does NOT go in the corresponding header section.


On Sat, Feb 18, 2023 at 4:54 AM SSwiniarski ***@***.***>
wrote:

> @easoftwareGit <https://github.com/easoftwareGit> so far so good. Now you
> need to edit the metadata so that the list of Subjects: are only from
> items that appear in the subjects.md file, and the items in Tags: only
> come from the tags.md file, and yo only use CatalogContent: as it appears
> in the catalog-content.md. Those files exist to show you what is
> acceptable content for those elements of the metadata.
>
> I want to make sure you understand this before you make any further edits.
> You use the files in docs/documentation/ to tell you what you can list in
> the metadata.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Codecademy/docs/pull/1627#issuecomment-1435663792>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALBP54WQ5JS4OZHX6WVULATWYDBCBANCNFSM6AAAAAAUV2K2HM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

SSwiniarski commented:
@easoftwareGit that's right.

easoftwareGit commented:
I have completed updating the Subjects, Tags and CatalogContent sections in
the header of qr-codes.md. Reading through the contents
of catalog-content.md, I could not find any entries that should be included
in qr-codes.md. Should I delete the CatalogContent section the the header
of qr-codes.md, or should I leave it blank?

On Sun, Feb 19, 2023 at 6:45 AM SSwiniarski ***@***.***>
wrote:

> @easoftwareGit <https://github.com/easoftwareGit> that's right.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Codecademy/docs/pull/1627#issuecomment-1436007610>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALBP54T5BJKNCGYHV43SC2DWYIWYZANCNFSM6AAAAAAUV2K2HM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

SSwiniarski commented:
@easoftwareGit use the following:

```
CatalogContent:
  - 'paths/code-foundations'
  - 'paths/computer-science'
```

SSwiniarski commented:
@easoftwareGit It's been two weeks, are you still working on this PR. All we need for the next review is the update to CatalogContent: I mentioned above.

easoftwareGit commented:
I added the two items to CatalogContent and committed the changes right
after you suggested them.

I was waiting for further instructions


On Sat, Mar 4, 2023 at 7:36 AM SSwiniarski ***@***.***> wrote:

> @easoftwareGit <https://github.com/easoftwareGit> It's been two weeks,
> are you still working on this PR. All we need for the next review is the
> update to CatalogContent: I mentioned above.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Codecademy/docs/pull/1627#issuecomment-1454779157>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALBP54ROBTWKGLRLCRNOWI3W2NOOZANCNFSM6AAAAAAUV2K2HM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

easoftwareGit commented:
I am out of town for a week. I did add the two suggested items
to CatalogContent and committed the changes.

On Sat, Mar 4, 2023 at 7:36 AM SSwiniarski ***@***.***> wrote:

> @easoftwareGit <https://github.com/easoftwareGit> It's been two weeks,
> are you still working on this PR. All we need for the next review is the
> update to CatalogContent: I mentioned above.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Codecademy/docs/pull/1627#issuecomment-1454779157>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALBP54ROBTWKGLRLCRNOWI3W2NOOZANCNFSM6AAAAAAUV2K2HM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

SSwiniarski commented on content/general/concepts/qr-code/qr-code.md:
```diff
@@ -0,0 +1,41 @@
+---
+Title: 'QR Codes'
+Description: 'QR Codes and how are they used'
```
The discription should be a description of the subject, not of the entry.

SSwiniarski reviewed (approved):
Aside from one comment, should be good for a second review.

KTom101 reviewed (changes requested):
@easoftwareGit I added a few changes to this. You will also need to remove the `.gitignore` it shouldn't be touched and there are no changes to the `subjects.md` file. You can also [fix the checks](https://github.com/Codecademy/docs/blob/main/.github/CONTRIBUTING.md#i-submitted-a-pr-but-tests-are-failing-how-do-i-fix-them) in the process. If you have trouble with the checks I can do them on my end just let me know. Please tag me when you're done.

easoftwareGit commented:
Resolved all the suggestions. I still have two questions

1) How do I remove the .gitignore file correctly? Do I just delete it, or
is there a way to flag it so it is not included?
2) Should I fix the checks AFTER I remove the .gitignore file?

On Mon, Mar 13, 2023 at 7:03 PM KTom101 ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> @easoftwareGit <https://github.com/easoftwareGit> I added a few changes
> to this. You will also need to remove the .gitignore it shouldn't be
> touched and there are no changes to the subjects.md file. You can also fix
> the checks
> <https://github.com/Codecademy/docs/blob/main/.github/CONTRIBUTING.md#i-submitted-a-pr-but-tests-are-failing-how-do-i-fix-them>
> in the process. If you have trouble with the checks I can do them on my end
> just let me know. Please tag me when you're done.
> ------------------------------
>
> In content/general/concepts/qr-code/qr-code.md
> <https://github.com/Codecademy/docs/pull/1627#discussion_r1134552990>:
>
> > +
> +Some QR Codes are static, and some are dynamic. A static QR Code has all the data encoded, and the data will not change. If a data entry error is made during QR Code creation, that error is permanent. A dynamic QR Code fixes the problem. A dynamic QR Code will retrieve data from the internet, allowing the QR Code owner to edit the data as needed.
> +
> +QR Codes do need to be square in shape, but there is room for some artistic styling. The squares and data dots can have rounded corners. The QR Code does not have to be black on white. The black can be changed to another color, but a higher contrast leads to more readability. A logo can be included in the center square. These changes make the QR Code visually unique and more suited to its use.
> +
> +## Reading a QR Code
> +
> +To read a QR Code one does not need a special laser bar code reader. Most camera apps in mobile devices contain the QR Code scanning software. Just open the camera app, focus on the QR Code, and a clickable link appears in the camera app.
> +
> +## Creating a QR Code
> +
> +Software is used to create the QR Code. The user enters data to be encoded and how the QR Code will be used. When the user is done entering the data, the software will create the QR Code. There are several QR Code generator websites. Some will generate basic QR Codes for free.
> +
> +## Uses for QR Codes
> +
> +QR Codes have many more uses than regular bar codes. A QR Codes hold more data, and they are easily readable by today’s mobile devices. The data can be just text, or instructions to open an app and populate fields. A QR Code can hold the URL for a web page or help log into a Wi-Fi network. QR Codes can show proof of purchase.  They are also used to track how many times the QR Code has been scanned. There are several other uses as well.
>
> ⬇️ Suggested change
>
> -QR Codes have many more uses than regular bar codes. A QR Codes hold more data, and they are easily readable by today’s mobile devices. The data can be just text, or instructions to open an app and populate fields. A QR Code can hold the URL for a web page or help log into a Wi-Fi network. QR Codes can show proof of purchase.  They are also used to track how many times the QR Code has been scanned. There are several other uses as well.
> +QR Codes have many more uses than regular bar codes. QR Codes hold more data, and they are easily readable by today’s mobile devices. The data can be just text, or instructions to open an app and populate fields. A QR Code can hold the URL for a web page or help log into a Wi-Fi network. QR Codes can show proof of purchase.  They are also used to track how many times the QR Code has been scanned. There are several other uses as well.
>
> ------------------------------
>
> In content/general/concepts/qr-code/qr-code.md
> <https://github.com/Codecademy/docs/pull/1627#discussion_r1134554695>:
>
> > +
> +QR Codes do need to be square in shape, but there is room for some artistic styling. The squares and data dots can have rounded corners. The QR Code does not have to be black on white. The black can be changed to another color, but a higher contrast leads to more readability. A logo can be included in the center square. These changes make the QR Code visually unique and more suited to its use.
> +
> +## Reading a QR Code
> +
> +To read a QR Code one does not need a special laser bar code reader. Most camera apps in mobile devices contain the QR Code scanning software. Just open the camera app, focus on the QR Code, and a clickable link appears in the camera app.
> +
> +## Creating a QR Code
> +
> +Software is used to create the QR Code. The user enters data to be encoded and how the QR Code will be used. When the user is done entering the data, the software will create the QR Code. There are several QR Code generator websites. Some will generate basic QR Codes for free.
> +
> +## Uses for QR Codes
> +
> +QR Codes have many more uses than regular bar codes. A QR Codes hold more data, and they are easily readable by today’s mobile devices. The data can be just text, or instructions to open an app and populate fields. A QR Code can hold the URL for a web page or help log into a Wi-Fi network. QR Codes can show proof of purchase.  They are also used to track how many times the QR Code has been scanned. There are several other uses as well.
> +
> +One of the most common uses for QR Codes is an URL for a web page. Instead of requiring a user to enter “codecademy.com/resources/docs/general” into a web browser, the URL is embedded into a QR Code. The user scans the QR Code and clicks the link. Many restaurants use QR Code URLs for on-line menus. Sporting events use QR Codes allowing spectators can get up to the minute stats. Charitable organizations use QR Codes directing doners to the donation web site.
>
> ⬇️ Suggested change
>
> -One of the most common uses for QR Codes is an URL for a web page. Instead of requiring a user to enter “codecademy.com/resources/docs/general” into a web browser, the URL is embedded into a QR Code. The user scans the QR Code and clicks the link. Many restaurants use QR Code URLs for on-line menus. Sporting events use QR Codes allowing spectators can get up to the minute stats. Charitable organizations use QR Codes directing doners to the donation web site.
> +One of the most common uses for QR Codes is an URL for a web page. Instead of requiring a user to enter “codecademy.com/resources/docs/general” into a web browser, the URL is embedded into a QR Code. The user scans the QR Code and clicks the link. Many restaurants use QR Code URLs for online menus. Sporting events use QR Codes allowing spectators to get up-to-the-minute stats. Charitable organizations use QR Codes to direct donors to the donation website.
>
> ------------------------------
>
> In content/general/concepts/qr-code/qr-code.md
> <https://github.com/Codecademy/docs/pull/1627#discussion_r1134555905>:
>
> > +
> +To read a QR Code one does not need a special laser bar code reader. Most camera apps in mobile devices contain the QR Code scanning software. Just open the camera app, focus on the QR Code, and a clickable link appears in the camera app.
> +
> +## Creating a QR Code
> +
> +Software is used to create the QR Code. The user enters data to be encoded and how the QR Code will be used. When the user is done entering the data, the software will create the QR Code. There are several QR Code generator websites. Some will generate basic QR Codes for free.
> +
> +## Uses for QR Codes
> +
> +QR Codes have many more uses than regular bar codes. A QR Codes hold more data, and they are easily readable by today’s mobile devices. The data can be just text, or instructions to open an app and populate fields. A QR Code can hold the URL for a web page or help log into a Wi-Fi network. QR Codes can show proof of purchase.  They are also used to track how many times the QR Code has been scanned. There are several other uses as well.
> +
> +One of the most common uses for QR Codes is an URL for a web page. Instead of requiring a user to enter “codecademy.com/resources/docs/general” into a web browser, the URL is embedded into a QR Code. The user scans the QR Code and clicks the link. Many restaurants use QR Code URLs for on-line menus. Sporting events use QR Codes allowing spectators can get up to the minute stats. Charitable organizations use QR Codes directing doners to the donation web site.
> +
> +QR Codes are also used as entry passes. When purchasing a movie ticket online, the user will be sent a QR Code. The user retrieves the QR Code on a mobile device, and at the movie theater, the user has the QR Code scanned and is admitted.
> +
> +But QR Codes can also be used to point a user to a malicious web site, just like a phishing email. One should be careful about scanning and clicking on a QR Code. A QR Code passed out at a conference, enabling the attendees to log into the local Wi-Fi is going to be safe to scan and click. On the other hand, a QR Code on a flyer stuck to a lamppost might be suspect. Just like emails, use good judgement before scanning and clicking.
>
> ⬇️ Suggested change
>
> -But QR Codes can also be used to point a user to a malicious web site, just like a phishing email. One should be careful about scanning and clicking on a QR Code. A QR Code passed out at a conference, enabling the attendees to log into the local Wi-Fi is going to be safe to scan and click. On the other hand, a QR Code on a flyer stuck to a lamppost might be suspect. Just like emails, use good judgement before scanning and clicking.
> +But QR Codes can also be used to point a user to a malicious web site, just like a phishing email. One should be careful about scanning and clicking on a QR Code. A QR Code passed out at a conference, enabling the attendees to log into the local Wi-Fi is going to be safe to scan and click. On the other hand, a QR Code on a flyer stuck to a lamppost might be suspect. Just like emails, use good judgment before scanning and clicking.
>
> ------------------------------
>
> In content/general/concepts/qr-code/qr-code.md
> <https://github.com/Codecademy/docs/pull/1627#discussion_r1134556433>:
>
> > +
> +A **QR Code** (quick response code) is code that can be read by a computer, just like the standard bar codes that appear on products at retail stores. But the QR Code is different than the standard bar code. The standard bar code is one dimensional and reads horizontally. A QR Code is a square, with the data enclosed within the square. The three squares in the corners of the QR code tell the QR Code scanning software how the QR Code is orientated, so the scanning software can read the QR Code’s data correctly. [Here](https://www.qr-code-generator.com/qr-code-marketing/qr-codes-basics/) is a good web page detailing how the data is structured in the QR Code.
> +
> +QR Codes are used just like their bar code counterparts: They contain data, and when scanned the data is entered into an app or is used by a device, greatly speeding up the data entry process and eliminating typing errors. Bar codes usually hold just a number, and once scanned and entered into an application. The scanned number can be used to lookup a specific item and retrieve the item’s name, price, and other information. QR Codes hold all kinds of data, so they are more widely used than bar codes.
> +
> +Some QR Codes are static, and some are dynamic. A static QR Code has all the data encoded, and the data will not change. If a data entry error is made during QR Code creation, that error is permanent. A dynamic QR Code fixes the problem. A dynamic QR Code will retrieve data from the internet, allowing the QR Code owner to edit the data as needed.
> +
> +QR Codes do need to be square in shape, but there is room for some artistic styling. The squares and data dots can have rounded corners. The QR Code does not have to be black on white. The black can be changed to another color, but a higher contrast leads to more readability. A logo can be included in the center square. These changes make the QR Code visually unique and more suited to its use.
> +
> +## Reading a QR Code
> +
> +To read a QR Code one does not need a special laser bar code reader. Most camera apps in mobile devices contain the QR Code scanning software. Just open the camera app, focus on the QR Code, and a clickable link appears in the camera app.
> +
> +## Creating a QR Code
> +
> +Software is used to create the QR Code. The user enters data to be encoded and how the QR Code will be used. When the user is done entering the data, the software will create the QR Code. There are several QR Code generator websites. Some will generate basic QR Codes for free.
>
> ⬇️ Suggested change
>
> -Software is used to create the QR Code. The user enters data to be encoded and how the QR Code will be used. When the user is done entering the data, the software will create the QR Code. There are several QR Code generator websites. Some will generate basic QR Codes for free.
> +Software is used to create the QR Code. The user enters the data to be encoded and how the QR Code will be used. When the user is done entering the data, the software will create the QR Code. There are several QR Code generator websites. Some will generate basic QR Codes for free.
>
> ------------------------------
>
> In content/general/concepts/qr-code/qr-code.md
> <https://github.com/Codecademy/docs/pull/1627#discussion_r1134556737>:
>
> > +  - 'paths/computer-science'
> +---
> +
> +## What is a QR Code?
> +
> +A **QR Code** (quick response code) is code that can be read by a computer, just like the standard bar codes that appear on products at retail stores. But the QR Code is different than the standard bar code. The standard bar code is one dimensional and reads horizontally. A QR Code is a square, with the data enclosed within the square. The three squares in the corners of the QR code tell the QR Code scanning software how the QR Code is orientated, so the scanning software can read the QR Code’s data correctly. [Here](https://www.qr-code-generator.com/qr-code-marketing/qr-codes-basics/) is a good web page detailing how the data is structured in the QR Code.
> +
> +QR Codes are used just like their bar code counterparts: They contain data, and when scanned the data is entered into an app or is used by a device, greatly speeding up the data entry process and eliminating typing errors. Bar codes usually hold just a number, and once scanned and entered into an application. The scanned number can be used to lookup a specific item and retrieve the item’s name, price, and other information. QR Codes hold all kinds of data, so they are more widely used than bar codes.
> +
> +Some QR Codes are static, and some are dynamic. A static QR Code has all the data encoded, and the data will not change. If a data entry error is made during QR Code creation, that error is permanent. A dynamic QR Code fixes the problem. A dynamic QR Code will retrieve data from the internet, allowing the QR Code owner to edit the data as needed.
> +
> +QR Codes do need to be square in shape, but there is room for some artistic styling. The squares and data dots can have rounded corners. The QR Code does not have to be black on white. The black can be changed to another color, but a higher contrast leads to more readability. A logo can be included in the center square. These changes make the QR Code visually unique and more suited to its use.
> +
> +## Reading a QR Code
> +
> +To read a QR Code one does not need a special laser bar code reader. Most camera apps in mobile devices contain the QR Code scanning software. Just open the camera app, focus on the QR Code, and a clickable link appears in the camera app.
>
> ⬇️ Suggested change
>
> -To read a QR Code one does not need a special laser bar code reader. Most camera apps in mobile devices contain the QR Code scanning software. Just open the camera app, focus on the QR Code, and a clickable link appears in the camera app.
> +To read a QR Code one does not need a special laser bar code reader. Most camera apps in mobile devices contain QR Code scanning software. Just open the camera app, focus on the QR Code, and a clickable link appears in the camera app.
>
> ------------------------------
>
> In content/general/concepts/qr-code/qr-code.md
> <https://github.com/Codecademy/docs/pull/1627#discussion_r1134798321>:
>
> > @@ -0,0 +1,41 @@
> +---
> +Title: 'QR Codes'
> +Description: 'QR Codes and how are they used'
> +Subjects:
> +  - Computer Science
> +Tags:
> +  - Images
> +CatalogContent:
> +  - 'paths/code-foundations'
> +  - 'paths/computer-science'
> +---
> +
> +## What is a QR Code?
> +
> +A **QR Code** (quick response code) is code that can be read by a computer, just like the standard bar codes that appear on products at retail stores. But the QR Code is different than the standard bar code. The standard bar code is one dimensional and reads horizontally. A QR Code is a square, with the data enclosed within the square. The three squares in the corners of the QR code tell the QR Code scanning software how the QR Code is orientated, so the scanning software can read the QR Code’s data correctly. [Here](https://www.qr-code-generator.com/qr-code-marketing/qr-codes-basics/) is a good web page detailing how the data is structured in the QR Code.
>
> ⬇️ Suggested change
>
> -A **QR Code** (quick response code) is code that can be read by a computer, just like the standard bar codes that appear on products at retail stores. But the QR Code is different than the standard bar code. The standard bar code is one dimensional and reads horizontally. A QR Code is a square, with the data enclosed within the square. The three squares in the corners of the QR code tell the QR Code scanning software how the QR Code is orientated, so the scanning software can read the QR Code’s data correctly. [Here](https://www.qr-code-generator.com/qr-code-marketing/qr-codes-basics/) is a good web page detailing how the data is structured in the QR Code.
> +A **QR Code** (quick response code) is code used to store information that can be read by a device with a QR scanner. QR Codes differ from barcodes seen in a retail or grocery store in that a standard barcode is one-dimensional and reads horizontally. A QR Code is square-shaped, with the data enclosed within the square. The three squares in the corners of the QR code tell the QR Code scanning software how the QR Code is orientated, so the scanning software can read the QR Code’s data correctly. [Here](https://www.qr-code-generator.com/qr-code-marketing/qr-codes-basics/) is a good web page detailing how the data is structured in the QR Code.
>
> ------------------------------
>
> In content/general/concepts/qr-code/qr-code.md
> <https://github.com/Codecademy/docs/pull/1627#discussion_r1134794085>:
>
> > @@ -0,0 +1,41 @@
> +---
> +Title: 'QR Codes'
> +Description: 'QR Codes and how are they used'
>
> The description still needs to be changed to be more about the actual
> subject.
> ⬇️ Suggested change
>
> -Description: 'QR Codes and how are they used'
> +Description: 'A QR Code, or quick response code, is code used to store information that can be read by a QR scanner.'
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Codecademy/docs/pull/1627#pullrequestreview-1337856622>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALBP54TJX4CBPXHFBP2RMTTW37GX7ANCNFSM6AAAAAAUV2K2HM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

KTom101 commented:
@easoftwareGit You can remove it from the PR by running 
`git checkout <branch> -- <file-name>` See the info linked [here](https://stackoverflow.com/questions/39459467/remove-a-modified-file-from-pull-request#:~:text=Switch%20to%20the%20branch%20from%20which%20you%20created%20the%20pull%20request%3A) for more details.

KTom101 commented:
@easoftwareGit Can you also make the changes I requested? You just marked them as resolved without adding them.

easoftwareGit commented:
Made the changes.
Got .gitignore removed
Did a commit.
Tried to create a Pull Request and got error message: GraphQL error: A pull
request already exists for easoftwareGit:easoftwareGit/issue282.

I don't have yarn installed, so I was not able to do the checking of the
format. KTom101 said they could do it on their end. Can you do it please?

On Tue, Mar 14, 2023 at 11:10 AM KTom101 ***@***.***> wrote:

> @easoftwareGit <https://github.com/easoftwareGit> Can you also make the
> changes I requested? You just marked them as resolved without adding them.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Codecademy/docs/pull/1627#issuecomment-1468593959>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALBP54VZJUYXDBOQ55BLFCLW4CYBVANCNFSM6AAAAAAUV2K2HM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

KTom101 commented:
@easoftwareGit Entry is [live](https://www.codecademy.com/resources/docs/general/qr-code) and you're currently listed as anonymous: 

![Screenshot 2023-03-18 at 8 59 33 PM](https://user-images.githubusercontent.com/40783324/226148131-832162a2-c4b3-46f5-b968-47ee9659efae.png)

If you want to be listed, follow the following steps:

1.) Add your GitHub Username to your Codecademy Profile Details, you can get there on your desktop by pressing on your profile picture in the upper right-hand corner and selecting "Profile" in the drop-down box:

<img width="871" alt="YourGitHubName" src="https://user-images.githubusercontent.com/40783324/226148188-71a5dfc5-39c6-4596-9243-293d53f271a6.png">

2.) Then log out and log back into Codecademy using your GitHub login:

<img width="1273" alt="ReLogIn" src="https://user-images.githubusercontent.com/40783324/226148210-ef9cc904-5194-4e6d-9bfc-a32b03906860.png">

> **Note:**  Your email used for Codecademy will have to be listed in your GitHub profile for the login feature to properly work. If you use the same email for both GitHub and Codecademy already, then you're set.

easoftwareGit commented:
I followed the steps provided by KTom101, and it worked! Saw the article
and my photo as one of the contributors!

I would like to thank all of you for your help on this article. As stated
before, I am dyslexic. I can structure a report, essay or article, I can
structure a paragraph, but it is hard for me to spell everything correctly,
and to catch mistakes that pass the spell checker but are still mistakes.
It is also hard for me to read and comprehend everything correctly, even
after multiple readings.

Thank you for your patience and guidance.

On Sat, Mar 18, 2023 at 6:07 PM KTom101 ***@***.***> wrote:

> @easoftwareGit <https://github.com/easoftwareGit> Entry is live
> <https://www.codecademy.com/resources/docs/general/qr-code> and you're
> currently listed as anonymous:
>
> [image: Screenshot 2023-03-18 at 8 59 33 PM]
> <https://user-images.githubusercontent.com/40783324/226148131-832162a2-c4b3-46f5-b968-47ee9659efae.png>
>
> If you want to be listed, follow the following steps:
>
> 1.) Add your GitHub Username to your Codecademy Profile Details, you can
> get there on your desktop by pressing on your profile picture in the upper
> right-hand corner and selecting "Profile" in the drop-down box:
>
> [image: YourGitHubName]
> <https://user-images.githubusercontent.com/40783324/226148188-71a5dfc5-39c6-4596-9243-293d53f271a6.png>
>
> 2.) Then log out and log back into Codecademy using your GitHub login:
>
> [image: ReLogIn]
> <https://user-images.githubusercontent.com/40783324/226148210-ef9cc904-5194-4e6d-9bfc-a32b03906860.png>
>
> *Note:* Your email used for Codecademy will have to be listed in your
> GitHub profile for the login feature to properly work. If you use the same
> email for both GitHub and Codecademy already, then you're set.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Codecademy/docs/pull/1627#issuecomment-1475057122>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALBP54THGOWEQIBURZDPNK3W4ZL5VANCNFSM6AAAAAAUV2K2HM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

SSwiniarski commented:
@easoftwareGit Well thank you for bearing with the process. We got a strong entry out of one of our oldest outstanding issues. You should be proud of it!

yangc95 commented:
@easoftwareGit Can we make an additional PR for this entry to edit the metadata? It should be a quick process as we'll consider this a typo.

On lines 5 & 7, let's go ahead and add in those single quotes. Thanks in advance!

> I can also assist with this change. Let me know how you would like to proceed.
